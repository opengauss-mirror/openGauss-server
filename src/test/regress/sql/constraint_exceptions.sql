-- 删除已有的表
DROP TABLE IF EXISTS products;

-- 创建产品表，设置唯一约束和检查约束
CREATE TABLE products (
    product_id SERIAL PRIMARY KEY,
    product_name VARCHAR(100) UNIQUE,
    price NUMERIC(10, 2) CHECK (price > 0)
);

-- 插入测试数据
INSERT INTO products (product_name, price) VALUES ('Laptop', 1000.00);

-- 尝试插入重复的产品名，期望违反唯一约束
INSERT INTO products (product_name, price) VALUES ('Laptop', 1200.00);

-- 尝试插入价格为负数的产品，期望违反检查约束
INSERT INTO products (product_name, price) VALUES ('Smartphone', -500.00);

-- 查询产品表，验证数据插入情况
SELECT product_name, price FROM products;
