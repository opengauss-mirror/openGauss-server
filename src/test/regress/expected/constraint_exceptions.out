ERROR:  duplicate key value violates unique constraint "products_product_name_key"
DETAIL:  Key (product_name)=(Laptop) already exists.

ERROR:  new row for relation "products" violates check constraint "products_price_check"
DETAIL:  Failing row contains (2, Smartphone, -500.00).

 product_name |  price
--------------+---------
 Laptop       | 1000.00
(1 row)
